spec/linkedlist.js

```diff
@@ -0,0 +1,195 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedlist'
+
+chai.use(chaiChange)
+
+const generateTestList = () => {
+  const list = new LinkedList()
+
+  list.insert('first')
+  list.insert('middle')
+  list.insert('last')
+
+  return list
+}
+
+
+describe('LinkedList', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  describe('insertFirst', () => {
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      expect(() => aLinkedList.insertFirst('foo'))
+        .to.alter(() => aLinkedList.size(), { from: 0, to: 1 })
+    })
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      aLinkedList.insertFirst('bar')
+
+      expect(() => aLinkedList[0]
+        .to.equal('bar'))
+    })
+  })
+
+  describe('insert()', () => {
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const aLinkedList = generateTestList()
```

Comment:
This test doesn't verify that the node was added to the tail, just that the size changed. Change the test description to reflect this.